MuscleModel/ReadFromModel/getMuscleInfo.m

```diff
@@ -1,154 +1,162 @@
+    for i=1:length(Misc.DofNames_muscles{t})
+        if ~any(strcmp(Misc.DofNames_Input{t},Misc.DofNames_muscles{t}{i}))
+             Inds_deleteDOFS(ct)=i;ct=ct+1;
+        end
+    end
+    Inds_deleteDOFS(ct:end)=[];
+    DOF_inds(Inds_deleteDOFS)=[];
+    Misc.DofNames{t}=Misc.DofNames_muscles{t}; Misc.DofNames{t}(Inds_deleteDOFS)=[];
+
+    % warnings when not all the input DOFS are actuated by muscles
+    for i=1:length(Misc.DofNames_Input{t})
+        if ~any(strcmp(Misc.DofNames_Input{t}{i},Misc.DofNames{t}))
+            disp(['Warning DOF: The input dof: ' Misc.DofNames_Input{t}{i} ' is not actuated by the selected muscles and therefore removed from the analysis']);
+        end
+    end
+
+    % Filter the moment arms information and store them in DatStore.dM
+    dM_raw=dM_temp(:,DOF_inds,:);
+    t_dM = dm_Data_temp.data(:,1);
+    fs=1/mean(diff(t_dM));
+    [B,A] = butter(Misc.f_order_dM, Misc.f_cutoff_dM/(fs/2));
+    DatStore(trial).dM = filtfilt(B,A,dM_raw);
+
+    % filter Muscle-tendon lengths and store them in DatStore.LMT
+    LMT_dat=importdata(fullfile(Misc.MuscleAnalysisPath,[Misc.MAtrialName{t} '_MuscleAnalysis_Length.sto']));
+    LMT_raw=LMT_dat.data(:,Inds_muscles);
+    t_lMT = LMT_dat.data(:,1);
+    fs=1/mean(diff(t_lMT));             % sampling frequency
+    [B,A] = butter(Misc.f_order_lMT,Misc.f_cutoff_lMT/(fs/2));
+    DatStore(trial).LMT = filtfilt(B,A,LMT_raw);
+
+    % store information in the DatStore structure
+    DatStore(trial).MuscleNames = Misc.MuscleNames{t};
+    DatStore(trial).DOFNames    = Misc.DofNames{t};
+    DatStore(trial).nMuscles    = length(Misc.MuscleNames{t});
+    DatStore(trial).nDOF        = length(Misc.DofNames{t});
+
+    %% Filter IK
+
+    % load joint kinematics
+    IK_data=importdata(IK_path);
+    if ~isfield(IK_data,'colheaders')
+        IK_data.colheaders=strsplit(IK_data.textdata{end});
     end
-end
-Inds_deleteDOFS(ct:end)=[];
-DOF_inds(Inds_deleteDOFS)=[];
-Misc.DofNames=Misc.DofNames_muscles; Misc.DofNames(Inds_deleteDOFS)=[];
-
-% warnings when not all the input DOFS are actuated by muscles
-for i=1:length(Misc.DofNames_Input)
-    if ~any(strcmp(Misc.DofNames_Input{i},Misc.DofNames))
-        disp(['Warning DOF: The input dof: ' Misc.DofNames_Input{i} ' is not actuated by the selected muscles and therefore removed from the analysis']);
+
+    % select the IK information between the selected time frames
+    t_IK=IK_data.data(:,1);
+    ind0=find(t_IK>=Misc.time(trial,1),1,'first'); ind_end=find(t_IK<=Misc.time(trial,2),1,'last');
+    IK_inds=ind0:ind_end;
+
+    % filter the kinematics and kinetics
+    fs=1/mean(diff(t_IK));
+    [B, A] = butter(Misc.f_order_IK, Misc.f_cutoff_IK/(fs/2));
+    [~,DOFS] = size(IK_data.data);
+    for i = 2:DOFS         % filtering time vector not needed (start from 2)
+        IK_data.data(:,i) = filtfilt(B,A,IK_data.data(:,i));
+    end
+
+    %% Filter ID
+    % Get the ID data
+    ID_data=importdata(ID_path);
```

Comment:
I think we want to get rid of importdata and want to use ReadMotFile because this is a lot faster !